text/2021-12-29-openapi-to-typescript-api-client-and-forms.md

```diff
@@ -0,0 +1,458 @@
+# OpenAPI to TypeScript API Client and Forms
+
+## Summary
+
+Use the OpenAPI specification to generate a TypeScript API client and forms.
+
+## Motivation
+
+Ref: https://github.com/chaos-mesh/chaos-mesh/issues/2615.
+
+Currently, the chaos types are split between the front-end and back-end, and we
+can‘t reuse types that have already been defined.
+
+You can find these **extra** type definitions in https://github.com/chaos-mesh/chaos-mesh/blob/release-2.1/ui/app/src/components/NewExperimentNext/data/types.ts.
+
+This leads to several problems:
+
+- The front-end needs to be manually synchronized when there is an update to the
+  type definition.
+- As `1` continues to emerge, there are more and more things to maintain manually.
```

Comment:
what is the `1` mean?

Reply:
Oops, I forgot to use a numbered list here, it should be:

1. The front-end needs to be manually synchronized when there is an update to the
  type definition.
2. As `1` continues to emerge, there are more and more things to maintain manually.